tests/shouldwork/AutoReg/AutoReg.hs

```diff
@@ -0,0 +1,127 @@
+{-|
+  Copyright   :  (C) 2019, Google Inc.
+  License     :  BSD2 (see the file LICENSE)
+  Maintainer  :  Christiaan Baaij <christiaan.baaij@gmail.com>
+-}
+{-# LANGUAGE CPP                   #-}
+{-# LANGUAGE PartialTypeSignatures #-}
+{-# LANGUAGE UndecidableInstances  #-}
+
+{-# OPTIONS_GHC -Wno-partial-type-signatures #-}
+{-# OPTIONS_GHC -fconstraint-solver-iterations=10 #-}
+
+-- Plugins don't load automatically when running the output tests
+{-# OPTIONS_GHC -fplugin=GHC.TypeLits.Extra.Solver #-}
+{-# OPTIONS_GHC -fplugin=GHC.TypeLits.Normalise #-}
+{-# OPTIONS_GHC -fplugin=GHC.TypeLits.KnownNat.Solver #-}
```

Comment:
Is it possible to use these flags with `runghc`?

Reply:
But because only this test required it, I thought at the time this would be better.
And actually because of other changes these aren't need here anymore. I forgot to take them out.
I'll do another commit to clean them up.